### DISCUSSION TITLE
Invalid data

### DISCUSSION THREAD

**Discussion by mrx2024** [aclap-dev/video-downloadhelper#98] [Help and Q&A]
### Your question

Hi, I used the plugin previously on a login site that I'm a member of and everything was fine before, but now I can't download anything without an error message coming up.

I'm on v8.2.0.23, mac 10.14.6 using Chrome. It asked me to update the companion app and I just did to v2.0.17.

Video DownloadHelper
Version 8.2.0.23
Browser locale: en-GB

Platform Mac x86-64
Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/116.0.0.0 Safari/537.36

The extension recognises the video but when I click to download, the same error message comes up that 'the host is down':

[tcp @ 0x7f892bf005c0] Connection to tcp://www.pianogroove.com:443 failed: Host is down
[in#0 @ 0x7f892be005c0] Error opening input: Host is down
Error opening input file https://www.pianogroove.com/?stream_loader=https%3A%2F%2Fpianogroove-com-encoded-vids-coconut-fra1.fra1.cdn.digitaloceanspaces.com%2FWebsite-Tour-Orientation%2FHome-Page-Explainer%2Findex.m3u8&pl=3&expire=1708159749&track=4938&signature=d43b61cca136c6c134b4bc9f252c6b57.
Error opening input files: Host is down

I've tried going to a few public sites to get a public link I can share with you but I can't find one that the plugin picks up on.

I've never had an error message before and everything's always run fine up until now.

**Comment by paulrouget** [comment]
> Addon & CoApp version and details
> No response

I'm missing this part.

We can't help you without that https://github.com/aclap-dev/video-downloadhelper/wiki/Where-can-I-find-my-versions%3F

---

Also, double checking, do you use a Mac antivirus or a firewall?

**Reply by GrampaWildWilly** [reply]
@paulrouget He's got that information in his post.  It's just not in the format you might be expecting.

**Reply by paulrouget** [reply]
No I don't know which version of the coapp he's using (there are multiple flavors of the same version).

**Reply by GrampaWildWilly** [reply]
Oh right.  He said he is on Mac 10.14.6.  That means he needs the other CoApp 2.0.17 specifically for that version of Mac.  I notice that his settings page that he posted a screenshot of below says it's verifying the CoApp.  Maybe he needs to get the other CoApp 2.0.17.

**Comment by GrampaWildWilly** [comment]
For what it's worth, I just went to your pianogroove site & it's up.  But like you said, it's a subscription site so I couldn't get in & look at anything.  Well, I clicked on something & it took me to this page:

https://www.pianogroove.com/join/

There happened to be a video on that page, a short teaser of 2:30.  Actually, there's a bunch of videos on that page but the teaser is the first one.  I managed to download the 1920x1080 variant of that without any problem.  I'm running Windows 11 Pro 64-bit version 23H2 build 22631.3155, Firefox 122.0.1 64-bit, licensed VDH 8.2.0.23a1 beta, CoApp 2.0.17.  Different operating system, different browser, possibly different VDH (not sure), same CoApp.  So this doesn't seem to be an issue with the site.  I don't know how representative of paid content this teaser might be so this test has a bunch of maybes & ifs with it.

Let's try something different.  Try this one:

https://www.medici.tv/en/operas/tchaikovsky-queen-of-spades-benedict-andrews-aziz-shokhakimov-asmik-grigorian-bayerische-staatsoper

That's also a membership site but that page has a universally available teaser of duration 0:36 on it.  Are you able to download that?

**Reply by mrx2024** [reply]
I get an error message when trying to download the free trailer video.

I also tried to download the link you sent but it doesn't recognise it:

![Screenshot 2024-02-15 at 10 38 51](https://github.com/aclap-dev/video-downloadhelper/assets/160113950/63e6ce97-c753-4ff8-8a4a-710b4f40de67)

**Comment by mrx2024** [comment]
Thank you for you fast responses. I can't copy the information to my clipboard so I took a screenshot.

![Screenshot 2024-02-15 at 10 29 57](https://github.com/aclap-dev/video-downloadhelper/assets/160113950/41bed2e4-ea31-4ba1-8488-7bcfee44ebda)

I do have a firewall but I have just enabled all Chrome processes. It helped in that it starts to run but now there's a different error message:

[hls @ 0x7fbf91504ac0] Error when loading first segment 'https://pianogroove-com-encoded-vids-coconut-fra1.fra1.cdn.digitaloceanspaces.com/Jazz-Piano-Foundations/1-Major-Scales/media-3/segment-0.ts?X-Amz-Algorithm=AWS4-HMAC-SHA256&X-Amz-Credential=KBGW2JT4XWKQ3NFEXFIX%2F20240215%2Fams3%2Fs3%2Faws4_request&X-Amz-Date=20240215T103621Z&X-Amz-Expires=14400&X-Amz-SignedHeaders=host&X-Amz-Signature=817f9720095ca95bb14fd479c87c45d8202e1bc555d72cf6626ffccbb47659af'
[in#0 @ 0x7fbf915042c0] Error opening input: Invalid data found when processing input
Error opening input file https://www.pianogroove.com/?stream_loader=https%3A%2F%2Fpianogroove-com-encoded-vids-coconut-fra1.fra1.cdn.digitaloceanspaces.com%2FJazz-Piano-Foundations%2F1-Major-Scales%2Findex.m3u8&pl=3&expire=1708166175&track=4938&signature=ad8da93098d560ba41b57fe0482055a7.
Error opening input files: Invalid data found when processing input

**Reply by paulrouget** [reply]
> I do have a firewall but I have just enabled all Chrome processes

Make sure it's disabled for the CoApp too.

**Reply by GrampaWildWilly** [reply]
Right.  Make exceptions in BOTH the antivirus AND the firewall for the ENTIRE CoApp directory.

**Reply by paulrouget** [reply]
In the submenu of the detected video, there's a "Details" section. Can you click on it, open the "core media" section, and tell me what's in there.

**Reply by GrampaWildWilly** [reply]
For the Medici opera teaser, I get this:

![#01](https://github.com/aclap-dev/video-downloadhelper/assets/75588964/85049cdd-a772-4550-95e5-f34983f00c35)
![#02](https://github.com/aclap-dev/video-downloadhelper/assets/75588964/c2f5a92e-085e-40b0-90f2-59d07895b786)
![#03](https://github.com/aclap-dev/video-downloadhelper/assets/75588964/9ac44823-8684-45a9-a3b8-fbbbdbb1b6d3)

I tried to tell Firefox to do a screenshot but it refused, saying it wasn't a normal web page.  Oh brother.  I tried putting the three screenshots into a single image but when I magnified it, I couldn't read it.  So I'm just posting the 3 screenshots.  They represent me scrolling through the core media.  I hope this is helpful.  They are tiny images here but you should be able to look at them in your browser & magnify them from there.

**Reply by mrx2024** [reply]
Ok, so I don't have anti-virus but for the firewall I enabled all permissions for the Coapp application and for Chrome.

I can't see a 'core media' section. I just see 'SideDownload Error' and I posted what's written in the last post. The error message is different since I enabled the permissions. First the message was 'host is down' and now it seems to say 'Invalid data found when processing input'

**Reply by mrx2024** [reply]
oh wow really? I must have missed it. Let me try searching for core media again.

**Reply by mrx2024** [reply]
I can't see core media but I see an option that says 'Details' and it looks a little similar. Is this helpful?

![Screenshot 2024-02-15 at 11 02 58](https://github.com/aclap-dev/video-downloadhelper/assets/160113950/abcbdaf8-f982-472a-8cb8-be73d62398e2)

**Reply by mrx2024** [reply]
Ah I'm sorry. I see it now.

**Reply by GrampaWildWilly** [reply]
When you say you permitted the CoApp in the firewall, what exactly did you permit?  The entire CoApp directory or just one executable?  I suggested the whole directory because there are multiple executables in there that need to be permitted.  It's just easier to make an exception for the directory.  Plus, you never know what Paul is going to come up with that will need to go into that directory at some future hypothetical date.  He's already added one new executable into that directory in the past few days.  I don't think it's something that needs to be permitted through the firewall, but the appearance of that executable made me realize he might come up with something some day that might need to go through the firewall.  So it's easier to just permit the whole directory now & you won't have to fiddle with it again.

**Reply by mrx2024** [reply]
Here's the 'core media' details:

![Screenshot 2024-02-15 at 11 06 04](https://github.com/aclap-dev/video-downloadhelper/assets/160113950/79162e0a-99e3-45c8-a551-8826d95e4acb)
![Screenshot 2024-02-15 at 11 06 15](https://github.com/aclap-dev/video-downloadhelper/assets/160113950/26da71cb-f2f5-4217-9238-88baa399f658)
![Screenshot 2024-02-15 at 11 06 27](https://github.com/aclap-dev/video-downloadhelper/assets/160113950/4efa05e9-825a-4377-bc47-21703d06b815)

**Reply by GrampaWildWilly** [reply]
In the Hit Details, there are 2 indicators that there is something hiding in there & you need to click it.  One is a little triangle & the other is {...}.  You need to click all of those to display the full core media.  There were a bunch of those when I did it, which is why it took 3 screenshots to capture them all.

**Reply by GrampaWildWilly** [reply]
Oops.  There's another.  Sometimes {} needs a click, too.

**Reply by mrx2024** [reply]
Thanks I think I just found it. It took me three screenshots too.

I use Little Snitch for my firewall. What I did was to enable all permissions for the application: net.downloadhelper.coapp for all servers. I also checked the network monitor and allowed permissions on anything that was running while the download was initiated.

**Reply by mrx2024** [reply]
My apologies. I think I got them all now:

![Screenshot 2024-02-15 at 11 14 26](https://github.com/aclap-dev/video-downloadhelper/assets/160113950/411b2c2a-9775-414f-bde5-0ca8357c9910)
![Screenshot 2024-02-15 at 11 14 38](https://github.com/aclap-dev/video-downloadhelper/assets/160113950/fd3b899e-905f-496b-ac20-00da70e96166)
![Screenshot 2024-02-15 at 11 14 53](https://github.com/aclap-dev/video-downloadhelper/assets/160113950/536ec58b-564a-4f74-bcc1-fec784f20a64)
![Screenshot 2024-02-15 at 11 15 06](https://github.com/aclap-dev/video-downloadhelper/assets/160113950/a42c9189-3ac6-49a2-a4fb-294bb9ce2785)

**Reply by GrampaWildWilly** [reply]
Here's what's in my CoApp directory:

C:\>cd "DownloadHelper CoApp"

C:\DownloadHelper CoApp>dir
 Volume in drive C is Windows 11
 Volume Serial Number is 2066-6331

 Directory of C:\DownloadHelper CoApp

02/08/2024  23:14    <DIR>          .
02/08/2024  07:14        49,135,616 ffmpeg.exe
02/08/2024  07:14        49,002,496 ffprobe.exe
02/08/2024  07:14         7,422,648 filepicker.exe
02/08/2024  07:14               468 google.json
02/08/2024  07:14               468 microsoft.json
02/08/2024  07:14               301 mozilla.json
02/08/2024  08:21            68,717 uninstall.exe
02/08/2024  07:14        38,888,401 vdhcoapp.exe
               8 File(s)    144,519,115 bytes
               1 Dir(s)  1,780,118,241,280 bytes free

The .exe files there show what you need to permit through your firewall.  As an example, the CoApp executes ffprobe at the start of its download.  That will appear, if it does at all, it usually runs very quickly, only at the start of the download, & then it won't appear again.  Honestly, you should just do the whole directory & be done with it.

**Reply by GrampaWildWilly** [reply]
@mrx2024 Did you notice the exchange above between Paul & me about which CoApp version you have?

**Reply by mrx2024** [reply]
I have a mac so I'm trying to see where any directories are. Usually I just have to allow permissions for the app. I did see ffmpeg and ffprobe in the network monitor and I allowed them.

No I think I missed that message. My Coapp version was working perfectly before today. It asked me to update it and I chose whichever version it showed for mac 10.14.

EDIT: There's only one installer I see for 10.13 & 10.14

**Reply by GrampaWildWilly** [reply]
I don't know anything about Mac but I think you got the right CoApp.  I think there's something in the stuff that's generated when you click the copy to clipboard button in the VDH settings that would tell Paul something significant.  I wouldn't recognize what it is but I'm sure he would.  I'm puzzled that the button isn't working for you.  I'm also puzzled by how it says in your screenshot of your VDH settings that it is verifying the CoApp.  That's especially puzzling since it says your license has been recognized.  I don't know enough to say for certain whether any of these things is significant.

net.downloadhelper.coapp . . .  Hmmm....  That used to be the name of the CoApp on Windows.  But with the advent of the 2.x CoApp versions, things changed.  The CoApp went into a different directory by default (I installed it in a directory I chose instead of letting it default), plus it got a new name:

![#05](https://github.com/aclap-dev/video-downloadhelper/assets/75588964/2b086a35-8a46-4ead-b6a9-68bd736fb75c)

I know it's not a great idea to draw analogies between Windows & Mac but still . . .

**Reply by mrx2024** [reply]
It might be because I'm on older OS but it's always come up as net.downloadhelper.coapp in the Applications folder:

![Screenshot 2024-02-16 at 00 28 51](https://github.com/aclap-dev/video-downloadhelper/assets/160113950/b4d91479-ace2-45c2-a02c-32024bfa4c47)

I just tried uninstalling the coApp and all files through CleanmyMacX and I reinstalled but its the same problem. It won't pick up certain formats and the ones it does, the process starts but an error always comes up.

Is it possible to try an earlier version? Or thinking outside the box, would it be worth trying it on another laptop to see the difference? Would I be able to use the licence on a newer laptop's Chrome and see if I can download to check whether its an issue on this laptop or with the new version of the coapp for the older OS? I have access to a newer macbook that doesn't have a firewall.

**Reply by GrampaWildWilly** [reply]
Licenses are for life & on whatever browser you bought it for on however many computers you own or will ever own.  You might end up locking your license but that is supposed to trigger an E-mail that tells you what to do.  I have recently had to do that myself & it worked fine, but that's for Firefox.  I can only hope it works the same for Chrome.

There is a repository of CoApps here: https://github.com/mi-g/vdhcoapp/releases/.  I think there are versions for older Macs on most of the recent versions.  As for the browser extension itself, the beta versions used to be kept here: https://www.downloadhelper.net/firefox/betas.  But beginning around 8.1.5.0a1 beta, that archive stopped being updated.  Plus that's for Firefox anyway.  There has never been an archive of backlevel VDH versions, neither general availability nor beta, for Chrome (nor Edge, but that's not relevant here).  The general availability & beta versions for Chrome (and Edge) have always been housed on the Google Chrome web store (& Microsoft Edge web store), & they don't keep archives.  For Firefox, the Mozilla add-on repository does have a few backlevel versions of the VDH browser add-on.  Those would all be what were general availability versions.  Clearly, that wouldn't include all the 8.2.x versions because those have all been betas.

N0.  Wait.  https://addons.mozilla.org/en-US/firefox/addon/video-downloadhelper/versions/.  That small archive has been removed.  In fact, I've just spent about a half hour doing searches for VDH on the Mozilla site & can't find VDH.  The VDH page is here, https://addons.mozilla.org/en-US/firefox/addon/video-downloadhelper/, but I found that by starting at https://www.downloadhelper.net/.  I think Mozilla might be shadow banning us.

Now, is it worth trying some backlevel software?  Beats me.  You know I don't work for VDH.  I'm just another user.  Plus I know Firefox on Windows & nothing else.  From where I sit, it's your call.  Paul may have other ideas.

After your removal & reinstall, is the name of the CoApp any different in that applications folder on your Mac?

**Reply by mrx2024** [reply]
Wow, thank you so much for all of your help. I feel like we're so close but I don't know what to do to get over the last hurdle.

The name stayed the same when I reinstalled. I followed your links and I tried installing two previous coapps and restarted the mac each time and then I tried adding the Chrome beta but it keeps showing the same error as the stable version - "Error when loading first segment"

Not sure what else to do at this point. Might try adding the licence to chrome on another laptop and see what happens.

**Reply by GrampaWildWilly** [reply]
There's a lot of generalized complaining in here about the current release.  You can see it if you read the most recent dozen or so discussions in the forum.  Maybe you're just hitting one of the problems that's supposed to be resolved in the next version.  By all appearances, Paul is feverishly trying to fix this.  It's probably worth trying that whenever he gets it finished & ready for us all to install.  I have to admit defeat here.  If I knew anything about Mac, I might have some brilliant ideas.  But I don't.  So you'll have to look for brilliance elsewhere.

**Reply by GrampaWildWilly** [reply]
If you're willing to use something other than VDH to get your content, at least for the moment, I might be able to help you with that.  I see in your Hit Details you posted above that the master manifest is an m3u8.  That means your pianogroove content is probably available as standard HLS streaming content.  I could suggest an alternative to VDH if you try this:

Go to one of your pianogroove pages & get the Hit Details on the highest resolution variant VDH displays.
Copy to your clipboard the URL shown in the masterManifest of the Hit Details.
Execute ffprobe providing it one parameter.  That would be the URL you just copied pasted from the clipboard.  That should look something like this:
ffprobe "URL"
The quotation marks are required in Windows for anything that contains such weird characters as : / ? & %, all of which appear in the URL in your image above.  I don't know how the Mac command processor differs from Windows.  You'd have to know this.  I'm assuming there's something in Mac for executing commands at a command prompt.  Do you call it Terminal?  Something like that?

There is an ffprobe & an ffmpeg included with the CoApp.  Let's try to use those & see if we can get it to work.  Execute that ffprobe with the URL as input & post the output it generates here.  We can take it from there.

**Reply by mrx2024** [reply]
I tried installing VDH and the licence on a new macbook but it said something about a disconnected port and it wouldn't let me download.

I followed your instructions but tbh its been a long time since I used terminal and it looks over my head right now. I guess I don't have much of a choice but to wait and see if a fix is coming at some point.

Much respect to you for trying to help out @GrampaWildWilly. It's much appreciated.

**Comment by paulrouget** [comment]
Is there a MKV option, or only MP4?

**Reply by GrampaWildWilly** [reply]
Didn't show any MKV for me.  Medici doesn't generally offer MKVs, only MP4s.

**Reply by mrx2024** [reply]
I only see mp4
